test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -599,6 +600,22 @@ private static String groupName(ThreadGroup threadGroup) {
     }
 
     /**
+     * Builds a set of count unique items. Make sure that the items properly implement equals and hashcode.
+     */
+    public static <T> Set<T> randomUnique(Supplier<T> supplier, int count) {
+        Set<T> things = new HashSet<>();
+        int maxTries = count * 10;
+        for (int t = 0; t < maxTries; t++) {
+            if (things.size() == count) {
+                return things;
+            }
+            things.add(supplier.get());
+        }
+        // Oh well, we didn't get enough unique things. It'll be ok.
```

Comment:
I think this is okay as long as you clearly put that constraint into the comment. Maybe also call "count" something more approximate, like "itemsRequested" or sth. and clearly state that this many items will only be returned if a) the supplier can output enough random objects and b) we are lucky enough to get that many. I think for most tests this should be fine.